prow/config/branch_protection_test.go

```diff
@@ -829,3 +832,196 @@ func TestConfig_GetBranchProtection(t *testing.T) {
 		})
 	}
 }
+
+func TestReposWithDisabledPolicy(t *testing.T) {
+	testCases := []struct {
+		name              string
+		config            Config
+		expectedRepoWarns []string
+	}{
+		{
+			name: "Warning is generated for repos with disabled policies",
+			config: Config{
+				ProwConfig: ProwConfig{
+					BranchProtection: BranchProtection{
+						Policy: Policy{
+							Protect: no,
+							RequiredStatusChecks: &ContextPolicy{
+								Contexts: []string{"hello", "world"},
+							},
+						},
+						AllowDisabledPolicies: true,
+						Orgs: map[string]Org{
+							"org1": {
+								Repos: map[string]Repo{
+									"repo1": {
+										Branches: map[string]Branch{
+											"branch1": {
+												Policy{
+													Protect: yes,
+												},
+											},
+										},
+									},
+									"repo2": {
+										Branches: map[string]Branch{
+											"branch1": {
+												Policy{
+													Protect: yes,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedRepoWarns: []string{"org1/repo1", "org1/repo2"},
+		},
+		{
+			name: "No warnings if disabled policies are not allowed",
```

Comment:
* No warnings if repo has no policies
* No warnings if repo's defined policy is protected

(same for branches below)

Reply:
Thanks. Added the test cases.